plugins/node/opentelemetry-instrumentation-bunyan/src/OpenTelemetryBunyanStream.ts

```diff
@@ -0,0 +1,147 @@
+ * limitations under the License.
+ */
+
+import { logs, SeverityNumber, Logger } from '@opentelemetry/api-logs';
+import type { LoggerOptions } from '@opentelemetry/api-logs';
+import type * as BunyanLogger from 'bunyan';
+import { VERSION } from './version';
+
+const DEFAULT_INSTRUMENTATION_SCOPE_NAME = 'io.opentelemetry.contrib.bunyan';
+const DEFAULT_INSTRUMENTATION_SCOPE_VERSION = VERSION;
+
+// This block is a copy (modulo code style and TypeScript types) of the Bunyan
+// code that defines log level value and names. These values won't ever change
+// in bunyan@1. This file is part of *instrumenting* Bunyan, so we want to
+// avoid a dependency on the library.
+const TRACE = 10;
+const DEBUG = 20;
+const INFO = 30;
+const WARN = 40;
+const ERROR = 50;
+const FATAL = 60;
+const levelFromName: { [name in BunyanLogger.LogLevelString]: number } = {
+  trace: TRACE,
+  debug: DEBUG,
+  info: INFO,
+  warn: WARN,
+  error: ERROR,
+  fatal: FATAL,
+};
+const nameFromLevel: { [level: number]: string } = {};
+Object.keys(levelFromName).forEach(function (name) {
+  nameFromLevel[levelFromName[name as BunyanLogger.LogLevelString]] = name;
+});
+
+const OTEL_SEV_NUM_FROM_BUNYAN_LEVEL: { [level: number]: number } = {
+  [TRACE]: SeverityNumber.TRACE,
+  [DEBUG]: SeverityNumber.DEBUG,
+  [INFO]: SeverityNumber.INFO,
+  [WARN]: SeverityNumber.WARN,
+  [ERROR]: SeverityNumber.ERROR,
+  [FATAL]: SeverityNumber.FATAL,
+};
+
+const EXTRA_SEV_NUMS = [
+  SeverityNumber.TRACE2,
+  SeverityNumber.TRACE3,
+  SeverityNumber.TRACE4,
+  SeverityNumber.DEBUG2,
+  SeverityNumber.DEBUG3,
+  SeverityNumber.DEBUG4,
+  SeverityNumber.INFO2,
+  SeverityNumber.INFO3,
+  SeverityNumber.INFO4,
+  SeverityNumber.WARN2,
+  SeverityNumber.WARN3,
+  SeverityNumber.WARN4,
+  SeverityNumber.ERROR2,
+  SeverityNumber.ERROR3,
+  SeverityNumber.ERROR4,
+  SeverityNumber.FATAL2,
+  SeverityNumber.FATAL3,
+  SeverityNumber.FATAL4,
+];
+
+function severityNumberFromBunyanLevel(lvl: number) {
+  // Fast common case: one of the known levels
+  const sev = OTEL_SEV_NUM_FROM_BUNYAN_LEVEL[lvl];
+  if (sev !== undefined) {
+    return sev;
+  }
+
+  // Otherwise, scale the Bunyan level range -- 10 (TRACE) to 70 (FATAL+10)
+  // -- onto the extra OTel severity numbers (TRACE2, TRACE3, ..., FATAL4).
+  // Values below bunyan.TRACE map to SeverityNumber.TRACE2, which is a bit
+  // weird.
+  return EXTRA_SEV_NUMS[
+    Math.min(
+      EXTRA_SEV_NUMS.length - 1,
+      Math.max(0, Math.floor(((lvl - 10) / (70 - 10)) * EXTRA_SEV_NUMS.length))
+    )
+  ];
+}
+
+/**
+ * A Bunyan stream for sending log records to the OpenTelemetry Logs Bridge API.
+ */
+export class OpenTelemetryBunyanStream {
+  private _otelLogger: Logger;
+
+  constructor(name?: string, version?: string, options?: LoggerOptions) {
+    this._otelLogger = logs.getLogger(
+      name || DEFAULT_INSTRUMENTATION_SCOPE_NAME,
+      version || DEFAULT_INSTRUMENTATION_SCOPE_VERSION,
+      options
+    );
```

Comment:
I'd prefer not to, at least not now. I think, as a user, I expect most instrumentations to produce trace and metric data. The Instrumentation instance has a `this._diag` for internal diagnostic messages on how the instrumentation is operating. I think I would *only* expect **log** data from the few instrumentations that are specifically for logging libraries.  For example, it would be surprising if `instrumentation-pg` resulted in `log` signal data.  (I'm not super confident in my expectations here, however.)

Reply:
Yes I think logs is a bit of a different case as I also wouldn't expect logs to be emitted by "normal" instrumentations, except for if the instrumentation is for a logging package like here. 

I think we should keep it like this for now (not having loggers in the instrumentation base) but if we see there are reasons that "normal" instrumentation would increasingly emit logs as well, we can add it to the instrumentation base.